src/labone/mock/hpk_mock_server.py

```diff
@@ -0,0 +1,137 @@
+from abc import ABC
+from contextlib import contextmanager
+import json
+
+from labone.core import ListNodesFlags, ListNodesInfoFlags
+from labone.core.helper import LabOneNodePath
+from labone.core.session import NodeInfo
+from labone.core.subscription import StreamingHandle
+from labone.core.value import AnnotatedValue
+from labone.mock.mock_server import ConcreteMockServer
+from labone.mock.hpk_functionality import HpkMockFunctionality
+
+
+HPK_SCHEMA_ID = 11970870220622790664
+SERVER_ERROR = "SERVER_ERROR"
+
+
+
+def build_capnp_error(error: Exception):
+    """
+    Returns:
+        Capnp Type dictionary for: Result(Error)
+    """
+    return {
+        "err": {
+            "code": 2,
+            "message": f"{error}",
+            "category": SERVER_ERROR,
+            "source": __name__,
+        }
+    }
+
+
+class HpkMockServer(ConcreteMockServer):
+    id = HPK_SCHEMA_ID
+    name = "HPK"
+
+    def __init__(self, functionality: HpkMockFunctionality) -> None:
+        super().__init__()
+        self._functionality = functionality
+
+    async def getTheSchema(self, _context, **kwargs):
+        """Redirect to server method."""
+        return await self._base_server.getTheSchema(_context, **kwargs)
```

Comment:
suggestion: Since all our servers need to implement the Reflection schema and in all cases we need to return the schema that was used to create the `MockServerImpl` I would move this into the `MockServerImpl` itself. This makes the workflow much safer and simpler to use. (Just pass the schema to `MockServerImpl`)